core/blockchain.go

```diff
@@ -1041,10 +914,10 @@ func (bc *BlockChain) LastConsensusAcceptedBlock() *types.Block {
 //
 // Note: During initialization, [acceptorTip] is equal to [lastAccepted].
 func (bc *BlockChain) LastAcceptedBlock() *types.Block {
```

Comment:
`CleanBlockRootsAboveLastAccepted` is a bit weird since it refetches the last accepted everytime rather than trying to clean the same block root when called, so I feel there can be a race.